staging/src/k8s.io/apiserver/pkg/util/flowcontrol/format/formatting.go

```diff
@@ -0,0 +1,237 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package format
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+
+	fcv1a1 "k8s.io/api/flowcontrol/v1alpha1"
+	"k8s.io/apiserver/pkg/authentication/user"
+	"k8s.io/apiserver/pkg/endpoints/request"
+)
+
+// This file provides an easy way to mark a value for printing in full
+// detail IF it is printed but without costing a lot of CPU or memory
+// if the value is NOT printed.  The API Priority and Fairness API
```

Comment:
Isn't just implementing GoString(), String(), or Format() on the types themselves a sufficient way to achieve that?

Reply:
I had lost track of the fact that String() or Format() could be used; those are better than GoString() because the latter abuses the format notation that calls for golang-syntax output.

Defining String() or Format() on the API types involves putting special code for that in the API package and fixes this behavior for all uses.  Would we want to do that?

Reply:
Switched to String() and `%s`.

Reply:
Yeah, if we don't want to change the general stringification of API objects or if it's contrary to local custom to put that code in the API package, then this formatting decorator is a reasonable approach. Thanks!